File: lingting-http/src/main/java/live/lingting/framework/http/api/ApiSimpleRequest.java
package live.lingting.framework.http.api;

import live.lingting.framework.http.HttpMethod;

import java.net.http.HttpRequest;

/**
 * @author lingting 2024-09-19 15:41
 */
public class ApiSimpleRequest extends ApiRequest {

	protected final HttpMethod method;

	protected final String uri;

	protected final HttpRequest.BodyPublisher body;

	public ApiSimpleRequest(HttpMethod method, String uri) {
		this(method, uri, null);
	}

	public ApiSimpleRequest(HttpMethod method, String uri, HttpRequest.BodyPublisher body) {
		this.method = method;
		this.uri = uri;
		this.body = body != null ? body : HttpRequest.BodyPublishers.noBody();
	}

	@Override
	public HttpMethod method() {
		return method;
	}

	@Override
	public String path() {
		return uri;
	}

	@Override
	public HttpRequest.BodyPublisher body() {
		return body;
	}

}

File: lingting-http/src/main/kotlin/live/lingting/framework/http/api/ApiSimpleRequest.kt
package live.lingting.framework.http.api

import live.lingting.framework.http.HttpMethod
import live.lingting.framework.http.body.Body

/**
 * @author lingting 2024-09-19 15:41
 */
open class ApiSimpleRequest @JvmOverloads constructor(
    protected val method: HttpMethod,
    protected val path: String,
    protected val body: Body = Body.empty()
) : ApiRequest() {
    override fun method(): HttpMethod {
        return method
    }

    override fun path(): String {
        return path
    }

    override fun body(): Body {
        return body
    }
}
